### PR TITLE
Utilize the 2to3 tool,and edit import library,so that  change origina…

### PR DESCRIPTION
**Pull request by N0nent1ty**
Hi, to whom it may concern: 
I just used the 2to3 python tool to make source code can be run in python3 environment, and change some part of grammar which use to be in python2 but no longer support by python3, like "file" function to "open" function, etc.

In the following days, I will also edit the "Readme file", add some instruction about how to install derrick and R-lang(environment setting).

but now the souce code able to be execute in python3 now, maybe you would like to have a check first.

Any question to this commit is welcome, below is my contact detail:
shengchikuo@gmail.com

### CODE REVIEW

File: pulsar/core/fuzzer.py [hunk 1]
```diff
@@ -409,10 +409,10 @@ def _get_constant_data(self):
         return 'A' * length
 
     def _get_non_utf8_data(self):
-        return "\xe7\x9f\x9b\xe7\x9b\u0178\xe4\u0153\xbf\
-                \xe4\u017e\u20ac\xe5\u20ac\xa7IT\xe5\xb7\
-                \u0161\xe5\u20ac\u017d\xe4\xb9\x8b\xe9\x97\
-                \u017d\xe4\xba\x89"
+        return "\xe7\x9f\x9b\xe7\x9b\\u0178\xe4\\u0153\xbf\
```
Comment (hgascon) [inline pulsar/core/fuzzer.py:412]: Why is this necessary?

File: pulsar/core/lens.py [hunk 2]
```diff
@@ -306,7 +306,7 @@ def find_similar_template(self, next_possible_states, message):
                 fields, dist = matcher.match()
                 templates += [(template, fields, distance)]
             sorted_templates = [(t, f, d) for t, f, d in sorted(templates,
-                                              key=lambda (t, f, d): (d, t, f))]
+                                              key=lambda t_f_d: (t_f_d[2], t_f_d[0], t_f_d[1]))]
```
Comment (hgascon) [inline pulsar/core/lens.py:309]: Is this part of python 3? Can you explain it a bit?

File: pulsar/core/sippy/UacStateRinging.py [hunk 3]
```diff
@@ -108,9 +108,9 @@ def recvEvent(self, event):
         #print 'wrong event %s in the Ringing state' % event
         return None
 
-if not globals().has_key('UaStateFailed'):
-    from UaStateFailed import UaStateFailed
-if not globals().has_key('UaStateConnected'):
-    from UaStateConnected import UaStateConnected
-if not globals().has_key('UacStateCancelling'):
-    from UacStateCancelling import UacStateCancelling
+if 'UaStateFailed' not in globals():
```
Comment (hgascon) [inline pulsar/core/sippy/UacStateRinging.py:111]: It is a bit strange that in some cases, "has_key" is changed for "in" and in other is the other way around, "in" is changed for "has_key". Do you have an explanation for that?

File: pulsar/core/sippy/SipTransactionManager.py [hunk 4]
```diff
@@ -22,27 +22,28 @@
...
+from .SipRequest import SipRequest
+from .SipAddress import SipAddress
+from .SipRoute import SipRoute
+from .SipHeader import SipHeader
 from datetime import datetime
 from hashlib import md5
 from traceback import print_exc
 from time import time
 import sys, socket
+from functools import reduce
```
Comment (hgascon) [inline pulsar/core/sippy/SipTransactionManager.py:38]: Why is this introduced?

File: pulsar/core/sippy/SipLogger.py [hunk 5]
```diff
@@ -22,12 +22,13 @@
...
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA.
 
-from Signal import Signal
+from .Signal import Signal
 from time import time, localtime, strftime
 from fcntl import flock, LOCK_EX, LOCK_UN
 from signal import SIGUSR1
 from errno import EINTR
 import sys, os
+from functools import reduce
```
Comment (hgascon) [inline pulsar/core/sippy/SipLogger.py:31]: Why is this introduced?

File: pulsar/core/sippy/SipCiscoGUID.py [hunk 6]
```diff
@@ -39,8 +39,8 @@ def __init__(self, body = None, ciscoGUID = None):
         if ciscoGUID != None:
             self.ciscoGUID = ciscoGUID
         else:
-            s = md5(str((random() * 1000000000L) + time())).hexdigest()
-            self.ciscoGUID = (long(s[0:8], 16), long(s[8:16], 16), long(s[16:24], 16), long(s[24:32], 16))
+            s = md5(str((random() * 1000000000) + time())).hexdigest()
+            self.ciscoGUID = (int(s[0:8], 16), int(s[8:16], 16), int(s[16:24], 16), int(s[24:32], 16))
```
Comment (hgascon) [inline pulsar/core/sippy/SipCiscoGUID.py:43]: I am not sure about this. Was this change introduced automatically?

**Review by hgascon — COMMENTED** [review]
Thank you very much for your PR.  Instead of updating the sippy modules here, it would make sense to review the current state of the original module at https://github.com/sippy/b2bua and update the whole package. Then indicate in the installation instructions where it can be downloaded from and where it needs to be put. How have you tested pulsar after the update?

File: pulsar/core/sippy/SipTransactionManager.py (hunk 4 above)
Comment (aszlig) [reply pulsar/core/sippy/SipTransactionManager.py:38]: `reduce` is [no longer a builtin in Python 3](https://docs.python.org/3.0/whatsnew/3.0.html#builtins):

> Removed reduce(). Use functools.reduce() if you really need it; however, 99 percent of the time an explicit for loop is more readable.

File: pulsar/core/sippy/SipCiscoGUID.py (hunk 6 above)
Comment (aszlig) [reply pulsar/core/sippy/SipCiscoGUID.py:43]: I'm not the author, but it was probably introduce automatically, since [int and long got unified in py3](https://www.python.org/dev/peps/pep-0237/).

File: pulsar/core/lens.py (hunk 2 above)
Comment (aszlig) [reply pulsar/core/lens.py:309]: The tuple unpacking syntax got removed, here is the list of reasons why:

https://www.python.org/dev/peps/pep-3113/#why-they-should-go
